blackjax/sgmcmc/sgnht.py

```diff
@@ -127,19 +127,25 @@ class sgnht:
     def __new__(  # type: ignore[misc]
         cls,
         grad_estimator: Callable,
+        alpha: float = 0.01,
+        beta: float = 0.0,
     ) -> MCMCSamplingAlgorithm:
-        kernel = cls.build_kernel()
-
-        def init_fn(position: PyTree, rng_key: PRNGKey):
-            return cls.init(rng_key, position)
-
-        def step_fn(rng_key: PRNGKey, state, minibatch: PyTree, step_size: float):
+        kernel = cls.build_kernel(alpha, beta)
+
+        def init_fn(
+            position: PyTree, rng_key: PRNGKey, init_xi: Union[None, float] = None
+        ):
+            return cls.init(position, rng_key, init_xi or alpha)
+
+        def step_fn(
+            rng_key: PRNGKey,
+            state: SGNHTState,
+            minibatch: PyTree,
+            step_size: float,
+            temperature: float = 1,
```

Comment:
As I understand it, SGNHT controls the temperature with `xi`, so we shouldn't need a temperature other than 1. If this is the case, we should at least add a `*,` before temperature so that the user that changes it knows what she is doing.

Reply:
I think this might be one case where the terminology is a bit confusing 😅 , my understanding from [Ding et al](https://proceedings.neurips.cc/paper/2014/file/21fe5b8ba755eeaece7a450849876228-Paper.pdf) is that `xi` automatically modifies the friction coefficient to force the "system temperature" to be near a "target temperature". This target temperature is the one defined above with default value 1, and behaves in the same way as in the other `sgmcmc` algorithms (just scales the logpdf/gradient).

Reply:
Actually looking at it now I think the temperature scaling might be incorrect in `diffusions.sghmc` and `diffusions.sgnht`

It should be either simply rescaling the gradient `g / temperature` or equivalently all `stepsize * temperature` right? Atm for `diffusions.sghmc` and `diffusions.sgnht` I think its doing the latter but incorrectly

Reply:
I see what you mean. Change was made in #396. It feels like `g / temperature` is the best option here.

Reply:
I think I actually prefer the `step_size * temperature` parameterization as it results in a nice transition to optimisation algorithms with `temperature` -> 0
SGLD -> SGD
SGHMC -> SGD with momentum
SGNHT -> something funky/cool/new 👀 

For SGHMC and SGNHT the temperature is applied to the augmented target distribution $\pi(x, v) \propto \exp(- (U(x) + \frac12 v^Tv) / T)$ rather than just the target in $x$

Probably this should be explained in the docs, and the transition to gradient descent(s) might make a nice little example.

All that needs to be changed is `step_size * beta` -> `step_size * temperature * beta` in `sghmc` and `sgnht` - I'll update the PR shortly.